rosbag2/src/rosbag2/writer.cpp

```diff
@@ -94,6 +120,23 @@ void Writer::write(std::shared_ptr<SerializedBagMessage> message)
     throw std::runtime_error("Bag is not open. Call open() before writing.");
   }
 
+  // Update the message count for the Topic.
+  ++topics_[message->topic_name].message_count;
+  ++metadata_.message_count;
```

Comment:
not that it matters, but I believe instead of constantly incrementing this message count, we could accumulate the number of messages in the `finalize_metadata` function when iterating through all topics.

Reply:
Oh we're not iterating over topics in `finalize_metadata`. Its iterating over `topics_`, which is an unordered map of topic name to `TopicInformation`.

Reply:
Yes, we iterate through `topics_` which in fact holds the message count, right? So the sum of all `message_count` of each item in the `topics_` should result in the overall message count. I think that's a bit more efficient than incrementing the `metadata_.message_count` with every call (I mean you do exactly that already in the line above).

Reply:
Oh I misinterpreted your question. Yeah I can do that.
I think it will be a wash when it comes to performance, but it does group together the operations if I do move it into `finalize_metadata`